src/rdf.rs

```diff
@@ -0,0 +1,275 @@
+        None,
+        None,
+    )
+}
+
+// Retrieve access urls of a distribution
+pub fn list_access_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::ACCESS_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve download urls of a distribution
+pub fn list_download_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::DOWNLOAD_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve distribution formats
+pub fn list_formats(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcterms::FORMAT).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve dataset namednode
+pub fn get_dataset_node(store: &Store) -> Option<NamedNode> {
+    list_datasets(&store).next().and_then(|d| match d {
+        Ok(Quad {
+            subject: Subject::NamedNode(n),
+            ..
+        }) => Some(n),
+        _ => None,
+    })
+}
+
+// Map GEO location method
+fn map_format_to_head(format_uri: String) -> String {
+    let fmt = format_uri.split("/").last().unwrap();
+    match fmt {
+        "WMS_SRVC" => "WMS".to_string(),
+        "WFS_SRVC" => "WFS".to_string(),
+        "WCS_SRVC" => "WCS".to_string(),
+        _ => "HEAD".to_string(),
+    }
+}
+
+// Extract accessURLs and downloadURLs from dataset
+pub fn extract_urls_from_distribution(
+    dist_node: NamedOrBlankNodeRef,
+    store: &Store,
+) -> Vec<HashMap<String, String>> {
+    let mut urls = Vec::new();
+
+    let mut url_extract = HashMap::new();
+    url_extract.insert("distribution_uri".to_string(), dist_node.to_string());
+
+    // Map format to HEAD
+    let head = list_formats(dist_node, store).next().map_or_else(
+        || "HEAD".to_string(),
+        |fmt| map_format_to_head(fmt.unwrap().to_string()),
+    );
+
+    for acc_url in list_access_urls(dist_node, store) {
+        match acc_url.unwrap().object {
+            Term::NamedNode(acc_url_node) => {
+                let mut new_url_extract = url_extract.clone();
+                new_url_extract.insert("method".to_string(), head.to_string());
+                new_url_extract.insert("access_url".to_string(), acc_url_node.into_string());
+                urls.push(new_url_extract);
+            }
+            node => error!("Access URL node is not a NamedNode but {}", node),
+        }
+    }
+
+    for dl_url in list_download_urls(dist_node, store) {
+        match dl_url.unwrap().object {
+            Term::NamedNode(dl_url_node) => {
+                let mut new_url_extract = url_extract.clone();
+                new_url_extract.insert("method".to_string(), head.to_string());
+                new_url_extract.insert("access_url".to_string(), dl_url_node.into_string());
```

Comment:
Skal dette være `download_url`?
```suggestion
                new_url_extract.insert("download_url".to_string(), dl_url_node.into_string());
```